src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/core/temperature/ShardStore.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.temperature;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.summaries.temperature.ShardProfileSummary;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import javax.annotation.Nonnull;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * The same set of shards will be seen for each metric and across multiple operations.
+ * This creates a pool of all shards so that they can be referenced from multiple places.
+ */
+public class ShardStore {
+    private static final Logger LOG = LogManager.getLogger(ShardStore.class);
+    /**
+     * The key for the outer map is indexName. The key for inner map is the ShardID. Given an
+     * indexName and shardId, a shard can be uniquely identified.
+     */
+    Map<String, Map<Integer, ShardProfileSummary>> list;
```

Comment:
Can we reuse Khushboo's IndexShardKey object as the key in this map ? The hashcode() in IndexShardKey is logically equivalent to the hashcode used in ElasticSearch to keep track of shards.

Reply:
That code is not merged yet. I will add a TODO to be addressed later. Thanks for the pointer.